agent/router/router.go

```diff
@@ -432,6 +458,27 @@ func (r *Router) HasDatacenter(dc string) bool {
 	return ok
 }
 
+func (r *Router) GetManager(areaID types.AreaID, dc string) *Manager {
+	r.RLock()
+	defer r.RUnlock()
+
+	area, ok := r.areas[areaID]
+	if !ok {
+		return nil
+	}
+
+	managerInfo, ok := area.managers[dc]
+	if !ok {
+		return nil
+	}
+
+	return managerInfo.manager
+}
+
+func (r *Router) GetLANManager() *Manager {
```

Comment:
All the codecov annotations complaining about coverage make this diff a bit hard to read. A simple happy-path test for `GetLANMananger` or `GetManager` might be good

Reply:
Removed GetManager and also added tests that will end up exercising GetLANManager